server/src/routes/opportunities/index.ts

```diff
@@ -106,59 +108,102 @@ const getStudentsStats = async (githubId: string) => {
   return studentStats.map((stats, idx) => ({ ...omit(stats, ['courseId']), ...studentPositions[idx] }));
 };
 
-export const getOpportunitiesConsent = (_: ILogger) => async (ctx: Router.RouterContext) => {
-  const { githubId } = ctx.query;
+const giveOpportunitiesConsent = async (ctx: Router.RouterContext, githubId: string) => {
+  const userRepository = getRepository(User);
 
-  const user = await getRepository(User).findOne({ where: { githubId } });
+  const creationResult = await createCV(githubId);
 
-  if (user === undefined) {
-    setResponse(ctx, NOT_FOUND);
+  if (creationResult === 'ALREADY_EXISTS') {
+    setResponse(ctx, CONFLICT, { message: 'CV already exists' });
     return;
   }
 
-  setResponse(ctx, OK, user.opportunitiesConsent);
-};
+  if (creationResult === 'CREATED') {
+    const result = await userRepository.update({ githubId }, { opportunitiesConsent: true });
+    setResponse(ctx, OK, result.raw.opportunitiesConsent);
+    return;
+  }
 
-export const setOpportunitiesConsent = (_: ILogger) => async (ctx: Router.RouterContext) => {
-  const { githubId } = ctx.request.body;
+  setResponse(ctx, INTERNAL_SERVER_ERROR, { message: 'CV creation failed' });
+};
 
+const withdrawOpportunitiesConsent = async (ctx: Router.RouterContext, githubId: string) => {
+  await deleteCV(githubId);
   const userRepository = getRepository(User);
-  const user = await userRepository.findOne({ where: { githubId } });
+  const result = await userRepository.update({ githubId }, { opportunitiesConsent: false });
+  setResponse(ctx, OK, result.raw.opportunitiesConsent);
+};
+
+export const manageOpportunitiesConsent = (_: ILogger) => async (ctx: Router.RouterContext) => {
+  const { githubId, opportunitiesConsent: reqConsent } = ctx.request.body;
```

Comment:
What does it mean? (reqConset)

Reply:
It means "requestedConsent". Removed for unnecessarity

Reply:
done